ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py

```diff
@@ -339,3 +339,78 @@ def should_connect_to_wan_rs(self: SharedUtils, path_groups: list) -> bool:
             local_path_group["name"] in path_groups and any(wan_interface["connected_to_pathfinder"] for wan_interface in local_path_group["interfaces"])
             for local_path_group in self.wan_local_path_groups
         )
+
+    @cached_property
+    def wan_ha(self: SharedUtils) -> bool:
+        """
+        Only trigger HA if 2 devices are in the same group and ha.enabled is true
+        """
+        if self.cv_pathfinder_role in [None, "pathfinder"]:
+            return False
+        return get(self.wan_site, "ha.enabled", default=True) and len(self.switch_data_node_group_nodes) == 2
+
+    @cached_property
+    def wan_ha_path_group_name(self) -> str:
+        """
+        Return HA path group name for the WAN design.
+
+        TODO make this configurable
+        """
+        return "LAN_HA"
+
+    @cached_property
+    def wan_ha_interfaces(self: SharedUtils) -> list:
+        """
+        Return the list of WAN HA interfaces
+        For now only picking up uplink interfaces in VRF default on the router.
+        """
+        # TODO probably does not need the whole uplink
+        return [uplink for uplink in self.get_switch_fact("uplinks") if get(uplink, "vrf") is None]
+
+    @cached_property
+    def wan_ha_peer_interfaces(self: SharedUtils) -> str:
+        return self.get_wan_peer_fact("wan_ha_interfaces")
+
+    @cached_property
+    def wan_ha_peer(self: SharedUtils) -> str | None:
+        # TODO - do we want a better mechanism here
+        if not self.wan_ha:
+            return None
+        if self.switch_data_node_group_nodes[0]["name"] == self.hostname:
+            return self.switch_data_node_group_nodes[1]["name"]
+        elif self.switch_data_node_group_nodes[1]["name"] == self.hostname:
+            return self.switch_data_node_group_nodes[0]["name"]
+        raise AristaAvdError("Unable to find WAN HA peer within same node group")
+
+    @cached_property
+    def wan_ha_peer_router_id(self: SharedUtils) -> str:
+        return self.get_wan_peer_fact("wan_ha_router_id")
+
+    @cached_property
+    def wan_ha_router_id(self: SharedUtils) -> str:
+        """
+        Get self.router_id from overlay.
+        """
+        return self.router_id
+
+    @cached_property
+    def wan_ha_ip_addresses(self: SharedUtils) -> list:
+        """
+        Return wan_ha_peer_interface ip_address
+        """
+        return [
+            get(interface, "ip_address", required=True, org_key=f"The uplink interface {interface} used as WAN LAN HA interface does not have an IP address")
+            for interface in self.wan_ha_interfaces
+        ]
+
+    @cached_property
+    def wan_ha_peer_ip_addresses(self: SharedUtils) -> list:
```

Comment:
Do we need all this, or could we just require an underlay routing protocol and assume the loopback0 to be reachable and peer on that?